Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodeMarkerDetector.cs

```diff
@@ -0,0 +1,257 @@
+// public facing development.
+// #define QRCODESTRACKER_BINARY_AVAILABLE
+
+#if QRCODESTRACKER_BINARY_AVAILABLE
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.QRCodesTracker;
+#endif
+
+using UnityEngine;
+
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection;
+using System.Collections.Generic;
+using System;
+
+#if UNITY_WSA && WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+using Windows.Perception.Spatial;
+using Windows.Perception.Spatial.Preview;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.MarkerDetection
+{
+    /// <summary>
+    /// QR code detector that implements <see cref="Microsoft.MixedReality.Toolkit.Extensions.MarkerDetection.IVariableSizeMarkerDetector"/>
+    /// </summary>
+    public class QRCodeMarkerDetector : MonoBehaviour,
+        IVariableSizeMarkerDetector
+    {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+        private QRCodesManager _qrCodesManager;
+        private Dictionary<Guid, SpatialCoordinateSystem> _markerCoordinateSystems = new Dictionary<Guid, SpatialCoordinateSystem>();
+#endif
+        private class QRCodeInfo
+        {
+            public Guid QRCodeId;
+            public int MarkerId;
+            public float MarkerSize;
+            public QRCodeInfo(Guid qrCodeId, int markerId, float markerSize)
+            {
+                this.QRCodeId = qrCodeId;
+                this.MarkerId = markerId;
+                this.MarkerSize = markerSize;
+            }
+        }
+
+        private bool _processMarkers = false;
+        private Dictionary<Guid, QRCodeInfo> _cachedMarkers = new Dictionary<Guid, QRCodeInfo>();
+        private Dictionary<int, List<Marker>> _markerObservations = new Dictionary<int, List<Marker>>();
+        private readonly string _qrCodeNamePrefix = "sv";
+
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+        private bool _tracking = false;
+#endif
+
+#pragma warning disable 67
+        /// <inheritdoc />
+        public event MarkersUpdatedHandler MarkersUpdated;
+#pragma warning restore 67
+
+        /// <inheritdoc />
+        public void SetMarkerSize(float markerSize){}
+
+        /// <inheritdoc />
+        public void StartDetecting()
+        {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+            _tracking = true;
+#else
+            Debug.LogError("Current platform does not support qr code marker detector");
+#endif
+        }
+
+        /// <inheritdoc />
+        public void StopDetecting()
+        {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+            _tracking = false;
+            _processMarkers = false;
+#else
+            Debug.LogError("Current platform does not support qr code marker detector");
+#endif
+        }
+
+        /// <inheritdoc />
+        public bool TryGetMarkerSize(int markerId, out float size)
+        {
+            bool found = false;
+            size = 0.0f;
```

Comment:
>size = 0.0f; [](start = 12, length = 12)

nit: I tend to not set the value upfront, and only set as close to the return statement or when I know for sure I have the value I intend to set. This way, I get a compile error if I by accident forget to set the value correctly before returning